### DISCUSSION TITLE
[Question] How to deserialize Kafka Stream with Avro?

### DISCUSSION THREAD

**Discussion by igoriok3001** [dotnet/spark#797] [Q&A]
I have a Kafka stream serialized with Avro, if you are using Scala or Java for Spark, then it is just a simple call to a function "from_avro".
More can be read here: https://databricks.com/blog/2018/11/30/apache-avro-as-a-built-in-data-source-in-apache-spark-2-4.html

How can I re-use "from_avro" in .net Spark?

Thanks.

**Comment by imback82** [comment]
Can you try this suggestion? https://github.com/dotnet/spark/discussions/592#discussioncomment-190407

Looks like we missed exposing avro related APIs and will expose them in coming releases.

cc @Niharikadutta

**Reply by Niharikadutta** [reply]
Sure, I will expose this.
